config/src/config/node_config_loader.rs

```diff
@@ -126,6 +126,18 @@ fn optimize_and_sanitize_node_config(
     NodeConfig::sanitize(node_config, node_type, chain_id)
 }
 
+pub fn sanitize_node_config(node_config: &mut NodeConfig) -> Result<(), Error> {
+    let node_type = NodeType::extract_from_config(node_config);
+    let chain_id = match get_chain_id(node_config) {
+        Ok(chain_id) => chain_id,
+        Err(error) => {
+            println!("Failed to get the chain ID from the genesis blob! Skipping config sanitization. Error: {:?}", error);
+            return Ok(());
+        },
+    };
+    NodeConfig::sanitize(node_config, node_type, chain_id)
```

Comment:
This sanitizes the testing configs?

Reply:
I think I found out that the sanitizer wasn't running because there was something in my overlay that it would have caught but instead I got a crash later (probably `None.unwrap()` for some `Option` in the config, but the sanitizer would have replaced None with a default value)

Reply:
Yeah, I originally never supported this path because I wasn't sure how useful it was. But, if you've found it useful, then it might be worth doing.

@bchocho: @brianolson is only calling `sanitize()` and not `optimize()` here, so it won't change any of the configs. Sanitize was meant to leave it unchanged 🤞

Reply:
The sanitizer is effectively where some defaults live, not in the value loaded by serde Default trait implementation just at/after the constructor, but later if something hasn't been set then fix it up at sanitizer time.